src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -144,11 +145,12 @@ public FileResource(FileService fileService, ResourceLoader resourceLoader, User
      * @param filename The filename of the file to get
      * @return The requested file, or 404 if the file doesn't exist
      */
-    @GetMapping("/files/templates/{filename:.+}")
-    public ResponseEntity<byte[]> getTemplateFile(@PathVariable String filename) {
+    @GetMapping({ "files/templates/{language}/{filename}", "/files/templates/{filename:.+}" })
+    public ResponseEntity<byte[]> getTemplateFile(@PathVariable Optional<ProgrammingLanguage> language, @PathVariable String filename) {
         log.debug("REST request to get file : {}", filename);
         try {
-            Resource fileResource = ResourcePatternUtils.getResourcePatternResolver(resourceLoader).getResource("classpath:templates" + File.separator + filename);
+            String languagePrefix = language.map(programmingLanguage -> File.separator + programmingLanguage.name().toLowerCase()).orElse("");
```

Comment:
So if there was no programmingLanguage selected, the languagePrefix will be "". Will this work?
I think we need to make this endpoint more specific now for the readme files (it is not used for anything else anyway) and make the language mandatory.

Reply:
I left it optional in case we will have no language specific files in the future. The question is, if this won't ever be used for something else than the readme files.

Reply:
Yes, for a prefix of "" the path just won't have the /java or /python folder, so you could get any other file from the templates folder

Reply:
I understand, for me this is optional and not a big problem. @maxr96 might disagree though and can offer some insight :)

Reply:
I think we should simply add `@PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")`
Then this won't be a big problem